homeassistant/components/envisalink.py

```diff
@@ -77,9 +78,21 @@
         vol.Optional(
             CONF_ZONEDUMP_INTERVAL,
             default=DEFAULT_ZONEDUMP_INTERVAL): vol.Coerce(int),
+        vol.Optional(
+            CONF_TIMEOUT,
+            default=DEFAULT_TIMEOUT): vol.Coerce(int),
     }),
 }, extra=vol.ALLOW_EXTRA)
 
+SERVICE_CUSTOM_FUNCTION = 'invoke_custom_function'
+CUSTOM_FUNCTION_ATTR = 'PGM'
```

Comment:
Service attribute constants should start with `ATTR_`.

The attribute string should be lowercase snakecase.